dashboard/admin.php

```diff
@@ -1,276 +1,312 @@
+                        $new_content .= $dom->saveHTML($body[$i]);
 
-            // This param in the database still using adsParams but in the frontend it's using customParams
-            if (!empty($params['ads_params']) && $params['ads_params'] !== null)
-                $dm_ce_data += ['ads_params' => self::sanitize_this('ads_params')];
+                        if ($i == $player_pos[0] - 1) {
+                            $new_content .= $player_holder['string'];
+                        }
+                    }
+                }
 
-            if (!empty($params['pre_video_title']) && $params['pre_video_title'] !== null)
-                $dm_ce_data += ['pre_video_title' => self::sanitize_this('pre_video_title')];
+            } else if (isset($player_holder['auto']) && $player_holder['auto'] === true) {
+                $new_content = '';
+                switch ($player_holder['pos']):
+            case 'top':
+                $new_content = $player_holder['string'] . $content;
+                break;
+            case 'middle':
+                $middle_pos = round(sizeof($body) / 2);
+                for ($i = 0; $i < sizeof($body); $i++) {
 
-            if (!empty($params['show_video_title']) && $params['show_video_title'] !== null)
-                $dm_ce_data += ['show_video_title' => self::sanitize_this('show_video_title')];
+                        $new_content .= $dom->saveHTML($body[$i]);
 
-            if (!empty($params['show_info_card']) && $params['show_info_card'] !== null)
-                $dm_ce_data += ['show_info_card' => self::sanitize_this('show_info_card')];
+                        if ($i == $middle_pos - 1) {
+                            $new_content .= $player_holder['string'];
+                        }
+                }
+                break;
+            default:
+                $new_content = $content . $player_holder['string'];
+                endswitch;
 
-            if (!empty($params['show_carousel_playlist']) && $params['show_carousel_playlist'] !== null)
-                $dm_ce_data += ['show_carousel_playlist' => self::sanitize_this('show_carousel_playlist')];
+            } else {
+                $new_content = $content;
+            }
 
-            if (!empty($params['mute']) && $params['mute'] !== null)
-                $dm_ce_data += ['mute' => self::sanitize_this('mute')];
+            return $new_content;
 
-            // Save the option to database
-            update_option('dm_ce_options_' . $tab, $dm_ce_data);
+        } // if the post and the page
 
-            echo '<div id="setting-error-settings_updated" class="notice notice-success settings-error is-dismissible"> 
-                    <p><strong>' . esc_html( __('Settings saved', 'dm_embed_plugin') ) . '</strong></p>
-                    </div>';
-        }
+        return $content;
     }
 
-    private function store_credentials($params)
+    public function migrate_old_player_to_custom_embed($content)
```

Comment:
@isatrio  this is the method to replace Old Dailymotion player into Custom Embed Player